content/pytorch/concepts/using-functional-apis/using-functional-api.md

```diff
@@ -0,0 +1,79 @@
+---
+Title: 'Using Functional APIs'
+Description: 'Functional APIs in PyTorch provide a flexible way to define and manipulate neural networks using functions rather than object-oriented classes.'
+Subjects:
+  - 'Machine Learning'
+  - 'Data Science' 
+Tags:
+  - 'PyTorch'
+  - 'Deep Learning'
+  - 'Neural Networks'
+CatalogContent:
+  - 'learn-python'
+  - 'paths/data-science'
+---
+
+**Functional APIs** in PyTorch provide a flexible and powerful way to define and manipulate neural networks. Unlike the `torch.nn.Module` class, which uses an object-oriented approach, functional APIs allow you to define models using functions. This can be particularly useful for creating complex models, experimenting with new architectures, or when you need more control over the forward pass.
+
+## Key Concepts
+
+### Functional Layers
+
+Functional layers are stateless and do not store parameters. Instead, parameters are passed explicitly. This can be useful for creating custom layers or when you need to reuse the same layer with different parameters.
+
+Example:
+
+```python
+import torch
+import torch.nn.functional as F
+
+# Define input tensor
+x = torch.randn(10, 3)
+
+# Define weights and bias
+weight = torch.randn(5, 3, requires_grad=True)
+bias = torch.randn(5, requires_grad=True)
+
+# Apply linear transformation
+output = F.linear(x, weight, bias)
+```
+
```

Comment:
Add output for the above code like this:

```suggestion
The code will output a tensor of shape `(10, 5)` representing the result of the linear transformation. However, the actual values of the tensor will be random due to the use of `torch.randn`, which generates random values from a normal distribution. 

The output might look like this:

```shell
tensor([[-1.1570, -1.0890, -0.4154, -0.1795,  1.6989],
        [-0.5629, -0.3360, -0.3411, -0.2352,  1.0300],
        [-0.3185,  0.2398,  0.5389,  0.2491,  1.0749],
        ...
        [ 0.0665,  0.4579, -0.1494, -0.5361,  0.7465],
        [-0.5970,  0.3147,  0.1569,  0.1582,  0.5355],
        [-0.4481, -0.5795,  0.4445, -0.0623,  0.7024]])
```
```